autoscale-core/src/main/java/com/hpe/caf/autoscale/core/ScalerThread.java

```diff
@@ -118,12 +118,10 @@ private void handleAnalysis()
             }
             governor.recordInstances(serviceRef, instances);
             ScalingAction action;
-            if (firstRun) {
-                LOG.debug("Performing initial scaling checks for service {}", serviceRef);
-                action = handleFirstRun(instances);
-                firstRun = false;
-            } else {
-                action = analyser.analyseWorkload(instances);
+            LOG.debug("Performing scaling checks for service {}", serviceRef);
+            action = analyser.analyseWorkload(instances);
+            if(action.getOperation() == ScalingOperation.NONE){
+                action = handleServiceInstanceLimits(instances);
```

Comment:
If we're no longer using `firstRun` then it should be removed.

Reply:
variable removed